### PR TITLE
Ft generic environment

### PR DESCRIPTION
**Pull request by maxspahn**
Adds generic parser for holonomic robots. 
It requires some 'ugly' changes in the extract_joint_ids because the urdf file has to be read before the center position of the robot is set if not specified by the user. Therefore, the read_limits is called before the environment reset function. 
But it must be called a second time to determine the correct indices of the controlled joints.

Open for suggestions on this one!

### CODE REVIEW

**Comment by c-salmi** [comment]
Did you add the generic_reacher to git? Running `python examples/generic_holonomic.py` gives 
image

**Comment by maxspahn** [comment]
Oh that's bad!!

**Comment by c-salmi** [comment]
Where should the `meshes` and `.urdf` live? Because they are now in `examples/` but for running `python examples/generic_holonomic.py` they need to be in the top level folder `/`. 

Maybe we could put them in [urdfenvs/generic_reacher/resources](https://github.com/maxspahn/gym_envs_urdf/tree/2b5594a2d9619a55d053d251ab60cc6b41739032/urdfenvs/generic_reacher/resources) like the other envs?

**Comment by maxspahn** [comment]
> Where should the meshes and .urdf live? Because they are now in examples/ but for running python examples/generic_holonomic.py they need to be in the top level folder /.

The whole point is the that user should be able to specify the location of the meshes. The location should be user-defined and set in the script that creates the environment.

> Maybe we could put them in [urdfenvs/generic_reacher/resources](https://github.com/maxspahn/gym_envs_urdf/tree/2b5594a2d9619a55d053d251ab60cc6b41739032/urdfenvs/generic_reacher/resources) like the other envs?`

The mesh for the ur5 should be an example to showcase how to integrate your own robot. Therefore, such meshes should not become part of the repository. Does this make sense?

**Comment by c-salmi** [comment]
Yes I agree that it should be user defined. Only I was trying to test with `python examples/generic_holonomic.py` but I should have first `cd examples`. I think it makes sense for the urdf to be in examples, so all good

**Review by c-salmi — APPROVED** [review]

File: urdfenvs/generic_reacher/resources/generic_robot.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import numpy as np
+from urdfenvs.urdfCommon.holonomic_robot import HolonomicRobot
+
+
+class GenericRobot(HolonomicRobot):
```
[the remark this answers is not in the document]
Comment (maxspahn) [reply urdfenvs/generic_reacher/resources/generic_robot.py:5]: I agree. I was alread in doubt when I was doing it.

File: examples/generic_holonomic.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+goal = False
+obstacles = False
+
+
+def main():
+    urdf_file = "ur5.urdf"
+
+    env = gym.make(
+        "generic-reacher-tor-v0", dt=0.01, urdf=urdf_file, render=True
```
[the remark this answers is not in the document]
Comment (maxspahn) [reply examples/generic_holonomic.py:13]: Yeah makes sense. Most torque environments are unstable to be honest.

**Pull request merged**